contrib/http_util/src/main/java/io/opencensus/contrib/http/util/HttpStatsUtil.java

```diff
@@ -0,0 +1,435 @@
+              Collections.<Double>unmodifiableList(
+                  Arrays.<Double>asList(
+                      0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 8.0, 10.0, 13.0, 16.0, 20.0, 25.0, 30.0,
+                      40.0, 50.0, 65.0, 80.0, 100.0, 130.0, 160.0, 200.0, 250.0, 300.0, 400.0,
+                      500.0, 650.0, 800.0, 1000.0, 2000.0, 5000.0, 10000.0, 20000.0, 50000.0,
+                      100000.0))));
+
+  /**
+   * {@link Measure} for the number of client-side HTTP requests started.
+   *
+   * @since 0.13
+   */
+  public static final MeasureLong HTTP_CLIENT_REQUEST_COUNT =
+      Measure.MeasureLong.create(
+          getName("client", "request_count"),
+          "Number of client-side HTTP requests started",
+          UNIT_COUNT);
+
+  /**
+   * {@link Measure} for the client-side HTTP request body size if set as ContentLength
+   * (uncompressed).
+   *
+   * @since 0.13
+   */
+  public static final MeasureLong HTTP_CLIENT_REQUEST_BYTES =
+      Measure.MeasureLong.create(
+          getName("client", "request_bytes"),
+          "Client-side HTTP request body size if set as ContentLength (uncompressed)",
+          UNIT_SIZE);
+
+  /**
+   * {@link Measure} for the client-side HTTP response body size (uncompressed).
+   *
+   * @since 0.13
+   */
+  public static final MeasureLong HTTP_CLIENT_RESPONSE_BYTES =
+      Measure.MeasureLong.create(
+          getName("client", "response_bytes"),
+          "Client-side HTTP response body size (uncompressed)",
+          UNIT_SIZE);
+
+  /**
+   * {@link Measure} for the client-side end-to-end latency.
+   *
+   * @since 0.13
+   */
+  public static final MeasureDouble HTTP_CLIENT_LATENCY =
+      Measure.MeasureDouble.create(
+          getName("client", "latency"), "Client-side end-to-end latency", UNIT_LATENCY);
+
+  /**
+   * {@link Measure} for the number of server-side HTTP requests started.
+   *
+   * @since 0.13
+   */
+  public static final MeasureLong HTTP_SERVER_REQUEST_COUNT =
+      Measure.MeasureLong.create(
+          getName("server", "request_count"),
+          "Number of server-side HTTP requests started",
+          UNIT_COUNT);
+
+  /**
+   * {@link Measure} for the server-side HTTP request body size if set as ContentLength
+   * (uncompressed).
+   *
+   * @since 0.13
+   */
+  public static final MeasureLong HTTP_SERVER_REQUEST_BYTES =
+      Measure.MeasureLong.create(
+          getName("server", "request_bytes"),
+          "Server-side HTTP request body size if set as ContentLength (uncompressed)",
+          UNIT_SIZE);
+
+  /**
+   * {@link Measure} for the server-side HTTP response body size (uncompressed).
+   *
+   * @since 0.13
+   */
+  public static final MeasureLong HTTP_SERVER_RESPONSE_BYTES =
+      Measure.MeasureLong.create(
+          getName("server", "response_bytes"),
+          "Server-side HTTP response body size (uncompressed)",
+          UNIT_SIZE);
+
+  /**
+   * {@link Measure} for the server-side end-to-end latency.
```

Comment:
Does "end-to-end" apply to server?

Reply:
Thanks! Maybe not. The specs for client latency is "Time between first byte of request headers sent to last byte of response received, or terminal error", so symmetrically the server latency would be something like "Time between first byte of request headers *received* to last byte of response *sent*, or terminal error". So compared to the client one, the server latency does not include time spend on receiving/sending data from/to client. So it is not a true e2e.

Reply:
Yes that's also what I was thinking - maybe we can just remove the word "end-to-end" here.